app/src/main/java/com/sjh/sunflower_sksowk156/MainActivity.kt

```diff
@@ -1,46 +1,177 @@
+import androidx.compose.runtime.MutableState
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.rememberCoroutineScope
+import androidx.compose.ui.Alignment
 import androidx.compose.ui.Modifier
+import androidx.compose.ui.geometry.Offset
+import androidx.compose.ui.input.nestedscroll.NestedScrollConnection
+import androidx.compose.ui.input.nestedscroll.NestedScrollSource
+import androidx.compose.ui.input.nestedscroll.nestedScroll
+import androidx.compose.ui.res.painterResource
 import androidx.compose.ui.tooling.preview.Preview
-import com.sjh.sunflower_sksowk156.ui.theme.Sunflowersksowk156Theme
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import com.sjh.sunflower_sksowk156.core.designsystem.theme.Sunflowersksowk156Theme
+import com.sjh.sunflower_sksowk156.feature.mygarden.MyGardenScreen
+import com.sjh.sunflower_sksowk156.feature.plantlist.PlantList
+import kotlinx.coroutines.launch
 
 class MainActivity : ComponentActivity() {
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContent {
             Sunflowersksowk156Theme {
-                // A surface container using the 'background' color from the theme
-                Surface(
-                    modifier = Modifier.fillMaxSize(),
-                    color = MaterialTheme.colorScheme.background
-                ) {
-                    Greeting("Android")
-                }
+                MainScreen()
             }
         }
     }
 }
 
+@OptIn(ExperimentalFoundationApi::class)
+@SuppressLint("UnusedMaterial3ScaffoldPaddingParameter")
 @Composable
-fun Greeting(name: String, modifier: Modifier = Modifier) {
-    Text(
-        text = "Hello $name!",
-        modifier = modifier
-    )
+fun MainScreen() {
+    val tabs = listOf("My garden", "Plant list")
+    val pagerState = rememberPagerState(pageCount = { tabs.size })
+    val showTopBar = remember { mutableStateOf(true) }
+
+    // NestedScrollConnection을 사용하여 스크롤에 따라 TopAppBar를 숨기거나 표시
+    val nestedScrollConnection = remember {
+        object : NestedScrollConnection {
+            override fun onPreScroll(available: Offset, source: NestedScrollSource): Offset {
+                // 스크롤 방향에 따라 TopAppBar의 보이기/숨기기 상태를 변경
+                val delta = available.y
+                showTopBar.value = delta > 0 || (delta == 0f && showTopBar.value)
+                return Offset.Zero
+            }
+        }
+    }
+    Scaffold(
+        modifier = Modifier.nestedScroll(nestedScrollConnection)
+    ) { paddingValues ->
+        Column(modifier = Modifier.fillMaxSize()) {
+            MainToolbar(showTopBar = showTopBar, currentPage = pagerState.currentPage)
+            MainContent(pagerState = pagerState, tabs = tabs)
+        }
+    }
+}
+
+@OptIn(ExperimentalFoundationApi::class)
+@Composable
+fun MainContent(tabs: List<String>, pagerState: PagerState) {
```

Comment:
`Tab(text = { Text(text = title) },`으로 미루어 볼 때, 조금 더 적절한 네이밍은
```suggestion
fun MainContent(tabTitles: List<String>, pagerState: PagerState) {
```
가 아닐까 생각합니다. 어떻게 생각하시나요?
`tabs`라는 변수명에 더 어울리는 타입은 `List<String>` 보다는 `List<TabOrWhateverSimilar>`일 것 같습니다.